synapse/config/oembed.py

```diff
@@ -58,7 +58,9 @@ def _parse_and_validate_providers(
         # Whether to use the packaged providers.json file.
         if not oembed_config.get("disable_default_providers") or False:
             providers = json.load(
-                pkg_resources.resource_stream("synapse", "res/providers.json")
+                importlib.resources.files("synapse")
+                .joinpath("res/providers.json")
+                .open("rb")
             )
```

Comment:
I think we probably want to use a context manager here to ensure the file gets closed, something like:

```python
providers_path = importlib.resources.files(...).joinpath(...)
with open(providers_path, "rb") as f:
    providers = json.load(f)
```